src/App.js

```diff
@@ -1,86 +1,48 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
+import { useLocation } from 'react-router-dom';
+import { NewTodo } from './components/NewTodo';
+import { FilterTodos } from './components/FilterTodos';
+import { TodoList } from './components/TodoList';
 
 function App() {
+  const [todos, setTodos] = useState([]);
+  const location = useLocation();
+  const { pathname } = location;
+
+  const filteredTodos = useMemo(() => {
+    let tempTodos = [...todos];
+
+    switch (pathname) {
+      case '/active':
+        tempTodos = todos.filter(todo => !todo.completed);
+        break;
+      case '/completed':
+        tempTodos = todos.filter(todo => todo.completed);
+        break;
+      default:
+        tempTodos = [...todos];
+        break;
```

Comment:
do you need to make a  copy of entire array, if there is no filter applied? Just return todos in this case